src/coreclr/vm/ilstubresolver.cpp

```diff
@@ -558,4 +558,44 @@ void ILStubResolver::StubGenFailed(ILStubResolver* pResolver)
     }
 
     pResolver->ClearCompileTimeState(ILNotYetGenerated);
-}
\ No newline at end of file
+}
+
+#ifndef DACCESS_COMPILE
+void FinalizeILStub(
+    _In_ ILStubResolver* ilResolver,
+    _In_ ILStubLinker* sl,
+    _Out_ DynamicResolver** dynamicResolver,
+    _Out_ COR_ILMETHOD_DECODER** methodILDecoder)
+{
+    STANDARD_VM_CONTRACT;
+    _ASSERTE(ilResolver != NULL);
+    _ASSERTE(sl != NULL);
+    _ASSERTE(dynamicResolver != NULL);
+    _ASSERTE(methodILDecoder != NULL);
+
+    UINT maxStack;
+    size_t cbCode = sl->Link(&maxStack);
+    DWORD cbSig = sl->GetLocalSigSize();
+
+    COR_ILMETHOD_DECODER* pILHeader = ilResolver->AllocGeneratedIL(cbCode, cbSig, maxStack);
+    BYTE* pbBuffer = (BYTE*)pILHeader->Code;
+    BYTE* pbLocalSig = (BYTE*)pILHeader->LocalVarSig;
+    _ASSERTE(cbSig == pILHeader->cbLocalVarSig);
+
+    size_t numEH = sl->GetNumEHClauses();
+    if (numEH > 0)
+    {
+        sl->WriteEHClauses(ilResolver->AllocEHSect(numEH));
+    }
+
+    sl->GenerateCode(pbBuffer, cbCode);
+    sl->GetLocalSig(pbLocalSig, cbSig);
+
+    // Store the token lookup map
+    ilResolver->SetTokenLookupMap(sl->GetTokenLookupMap());
+    ilResolver->SetJitFlags(CORJIT_FLAGS(CORJIT_FLAGS::CORJIT_FLAG_IL_STUB));
+
+    *dynamicResolver = (DynamicResolver*)ilResolver;
```

Comment:
```suggestion
    *dynamicResolver = ilResolver;
```
Should we change `class ILStubResolver : DynamicResolver` to `class ILStubResolver : public DynamicResolver` so that this cast is not needed?

Reply:
> out argument are not needed?

It is a bit hard to follow that this assignment transfers the ownership, but the other part of the ownership transfer (e.g. `ilResolver.SuppressRelease`) is in the caller. I think it would look better if the whole ownership transfer is done in one spot.

Reply:
Yes, ugh.

Reply:
> > out argument are not needed?
> 
> It is a bit hard to follow that this assignment transfers the ownership, but the other part of the ownership transfer (e.g. `ilResolver.SuppressRelease`) is in the caller. I think it would look better if the whole ownership transfer is done in one spot.

Yep, good point. Fixed.